File: src/test/scala/careercup/chapter1/StrangeReplacerTest.scala
package careercup.chapter1

import org.scalatest.FunSuite
import org.junit.Assert._

/**
 * Created by ahalikov on 24/04/15.
 */
class StrangeReplacerTest extends FunSuite {
  test("replace test") {
    assertEquals("", StrangeReplacer.replace(""))
    assertEquals("%20", StrangeReplacer.replace(" "))
  }
}

File: src/test/java/careercup/chapter1/StrangeReplacerTest.java
package careercup.chapter1;

import org.junit.Test;

import static careercup.chapter1.StrangeReplacer.*;
import static org.junit.Assert.*;

/**
 * @author ahalikov
 */
public class StrangeReplacerTest {

    @Test
    public void testReplaceString() throws Exception {
        assertEquals("", replaceString(""));
        assertEquals("%20", replaceString(" "));
        assertEquals("ab%20%20%20c%20%20d", replaceString("ab   c  d"));
    }

    @Test
    public void testReplaceChars() throws Exception {
        assertArrayEquals(null, replaceChars(null));
        assertArrayEquals(new char[]{}, replaceChars(new char[]{}));
        assertArrayEquals("a%20b".toCharArray(), replaceChars("a b".toCharArray()));
        assertArrayEquals("ab%20%20%20c%20%20d".toCharArray(), replaceChars("ab   c  d".toCharArray()));
    }
}
